pygmt/datasets/tile_map.py

```diff
@@ -0,0 +1,140 @@
+"""
+Function to load raster tile maps from XYZ tile providers, and load as
+:class:`xarray.DataArray`.
+"""
+
+try:
+    import contextily
+except ImportError:
+    contextily = None
+
+import numpy as np
+import xarray as xr
+
+__doctest_requires__ = {("load_tile_map"): ["contextily"]}
+
+
+def load_tile_map(region, zoom="auto", source=None, lonlat=True, wait=0, max_retries=2):
+    """
+    Load a georeferenced raster tile map from XYZ tile providers.
+
+    The tiles that compose the map are merged and georeferenced into an
+    :class:`xarray.DataArray` image with 3 bands (RGB). Note that the returned
+    image is in a Spherical Mercator (EPSG:3857) coordinate reference system.
+
+    Parameters
+    ----------
+    region : list
+        The bounding box of the map in the form of a list [*xmin*, *xmax*,
+        *ymin*, *ymax*]. These coordinates should be in longitude/latitude if
+        ``lonlat=True`` or Spherical Mercator (EPSG:3857) if ``lonlat=False``.
+
+    zoom : int
+        Optional. Level of detail. [Default is 'auto'].
```

Comment:
Hm, from the default it seems like that there is also string input possible. 
Currently, I am wondering what "level of detail" means. Do users have to pass a percentage value to `zoom`? Or is `zoom` something similar we have for `rivers` of `pygmt.Figure.coast`? Maybe we should also state what `"auto"` actually means?
```suggestion
    zoom : int or str
        Optional. Level of detail [Default is ``"auto"``].
```

Reply:
Good catch. Yes, I did think wonder if people actually knew what to use for `zoom`! The `zoom` here refers to an integer from 0 to 22, where 0 is the most zoomed out (i.e. the whole globe), and 22 is the most zoomed in (i.e. you can see buildings. See https://docs.mapbox.com/help/glossary/zoom-level or https://leafletjs.com/examples/zoom-levels for a good definition.

There is a small complication in that not all tile sources supports zoom up to level 22, so it's a bit hard to just document something like "Use an integer between 0 and 22 (inclusive)". But let me try adding a better description of this `zoom` parameter.

Reply:
Ok, more detail added in 9f14d31608f29c1aa8277b3c9b52470724eb7fd1. Let me know if that reads ok!

Reply:
Agree that the description of `zoom` should be expanded a little bit since in my opinion this is a great (GMT-) improvement for people doing work on small scale levels such as cities or communities and want to have geographic content (streets, buidldings, parks etc.) in the background as reference. I missed such things in the past and used other libraries or created customized stuff. Definitely a feature we should highlight in the next release! Great work @weiji14 :wink:

Reply:
Thanks @weiji14 for expanding the docstring for ``zoom``! Now it is clearer how to use this parameter 🙂.